x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -90,11 +92,16 @@ class RollupShardIndexer {
     private final String[] metricFields;
     private final String[] labelFields;
     private final Map<String, FieldValueFetcher> fieldValueFetchers;
-    private final AtomicLong numSent = new AtomicLong();
-    private final AtomicLong numIndexed = new AtomicLong();
-    private final AtomicLong numFailed = new AtomicLong();
+    private final AtomicLong numReceived;
+    private final AtomicLong numSent;
+    private final AtomicLong numIndexed;
+    private final AtomicLong numFailed;
+
+    private final RollupShardTask task;
+    private volatile boolean abort = false;
```

Comment:
If we set `abort` then the task cancel field isn't set to true. I think this field and task cancel field are very overlapping in term of usage. I think we should remove this field and use task cancel only here.

Reply:
because the `task.cancel()` is not a public method, so I add a new variable to do the abort logic internal.
The abort reason is from the bulk failed, so the RollupShardIndexer will throw exception. 
It's not from a user call to the cancel API.

Reply:
Ah, yes. I forgot about that. Yes, then the `abort` field makes sense here.